src/Tribe/Assets.php

```diff
@@ -207,46 +207,52 @@ public function enqueue( $forcibly_enqueue = null ) {
 	 * If the file does not exist, returns false.
 	 *
 	 * @since 4.3
+	 * @since TBD Removed ability to pass a filepath as $url
 	 *
-	 * @param string $url   The path or URL to the un-minified file.
+	 * @param string $url The absolute URL to the un-minified file.
 	 *
-	 * @return string|false The path/url to minified version or false, if file not found.
+	 * @return string|false The url to the minified version or false, if file not found.
 	 */
 	public static function maybe_get_min_file( $url ) {
 		$urls = array();
-		// If need add the Min Files
-		if ( ! defined( 'SCRIPT_DEBUG' ) || SCRIPT_DEBUG === false ) {
-			if ( substr( $url, - 3, 3 ) === '.js' ) {
-				$urls[] = substr_replace( $url, '.min', - 3, 0 );
-			}
 
-			if ( substr( $url, - 4, 4 ) === '.css' ) {
-				$urls[] = substr_replace( $url, '.min', - 4, 0 );
-			}
+		if ( 0 === strpos( $url, WPMU_PLUGIN_URL ) ) {
+			// URL inside WPMU plugin dir.
+			$plugins_dir = wp_normalize_path( WPMU_PLUGIN_DIR );
+			$base_url   = WPMU_PLUGIN_URL;
+		} elseif ( 0 === strpos( $url, WP_PLUGIN_URL ) ) {
+			// URL inside WP plugin dir.
+			$plugins_dir = wp_normalize_path( WP_PLUGIN_DIR );
+			$base_url   = WP_PLUGIN_URL;
+		} else {
+			// Resource needs to be inside a plugins dir, if not we can't check if it exists.
```

Comment:
I am not an enormous fan of this method anyways. It does things a bit backwards.

It asks that you convert a filepath to a URL and pass it here, we then immediately try to reverse engineer the URL and determine the original filepath, then we check if the file exists. But accurately reversing the URL is not easy due to all the possible configurations of WordPress, in fact it's impossible to do outside of certain directories.

A much better method would be one that asks you for a filepath, and then returns the possibly minified URL. This is easy, WordPress provides an API for it. Best of all it would be trivial to rejigger the `Tribe__Assets` to make full use of this, as `Tribe__Assets` converts from a filepath to a URL immediately before calling this method. And `Tribe__Assets` is the only place we even use this method.

IMO what we should do is ticket a new method that does what I am describing. We don't even have to deprecate this one, we could just stop using it. It's static and harmless, just limited in functionality due to what it's trying to do. What do you think @borkweb ? In the mean time can I merge this PR, which has the same limitation as the previous code in this regard.